x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/scalar/string/LengthTests.java

```diff
@@ -41,19 +41,28 @@ public static Iterable<Object[]> parameters() {
                 equalTo(UnicodeUtil.codePointCount(value))
             );
         }),
-            new TestCaseSupplier("empty string", () -> makeTestCase("", 0)),
-            new TestCaseSupplier("single ascii character", () -> makeTestCase("a", 1)),
-            new TestCaseSupplier("ascii string", () -> makeTestCase("clump", 5)),
-            new TestCaseSupplier("3 bytes, 1 code point", () -> makeTestCase("☕", 1)),
-            new TestCaseSupplier("6 bytes, 2 code points", () -> makeTestCase("❗️", 2)),
-            new TestCaseSupplier("100 random alpha", () -> makeTestCase(randomAlphaOfLength(100), 100)),
-            new TestCaseSupplier("100 random code points", () -> makeTestCase(randomUnicodeOfCodepointLength(100), 100))
+            new TestCaseSupplier("empty string", () -> makeTestCaseAsKeyword("", 0)),
+            new TestCaseSupplier("single ascii character", () -> makeTestCaseAsKeyword("a", 1)),
+            new TestCaseSupplier("ascii string", () -> makeTestCaseAsKeyword("clump", 5)),
+            new TestCaseSupplier("3 bytes, 1 code point", () -> makeTestCaseAsKeyword("☕", 1)),
+            new TestCaseSupplier("6 bytes, 2 code points", () -> makeTestCaseAsKeyword("❗️", 2)),
+            new TestCaseSupplier("100 random alpha", () -> makeTestCaseAsKeyword(randomAlphaOfLength(100), 100)),
+            new TestCaseSupplier("100 random code points", () -> makeTestCaseAsKeyword(randomUnicodeOfCodepointLength(100), 100)),
+            new TestCaseSupplier("ascii string as text", () -> makeTestCaseAsText("clump", 5))
```

Comment:
I suppose we could make both text and keyword in the function and return both.

It tends to be easier to do:
```
List<TestCase> cases = new ArrayList<>();
cases.addAll(makeTestCases("empty string", 0,  () -> ""));
cases.addAll(makeTestCases("100 random alpha", 100, () -> randomAlphaOfLength(100)));
```

I used to do with the `List.of` but for these it's just a pain.

Reply:
Fixed by https://github.com/elastic/elasticsearch/pull/103686/commits/cf5061ad8cefe2be86cfcb262748604091d7d673 . Thanks @luigidellaquila for support and patience 😅 .